eng/scripts/automation/GenerateAndBuildLib.ps1

```diff
@@ -366,15 +366,19 @@ function CreateOrUpdateCadlConfigFile() {
         [string]$directory,
         [string]$commit = "",
         [string]$repo = "",
-        [string]$specRoot = ""
+        [string]$specRoot = "",
+        [string]$additionalSubDirectories="" #additional directories needed, separated by semicolon if more than one
+        
     )
     if (!(Test-Path -Path $cadlConfigurationFile)) {
         New-Item -Path $cadlConfigurationFile
     }
 
+    Install-ModuleIfNotInstalled "powershell-yaml" "0.4.1" | Import-Module
```

Comment:
This install check is done three times in the script. Is it better to consolidate it, or is it done per-function to optimize for how the library functions are called?

Reply:
Thanks @benbp  This script file defines the functions which will be called in other executable script file.  The function need to contain everything it needed inside, otherwise, it may need the caller to separately set up those needed libraries before call this function which may not convenience and may cause failure.